polkadot/runtime/parachains/src/inclusion/mod.rs

```diff
@@ -935,6 +935,15 @@ impl<T: Config> Pallet<T> {
 		para: ParaId,
 		upward_messages: &[UpwardMessage],
 	) -> Result<(), UmpAcceptanceCheckErr> {
+		// Filter any pending UMP signals and the separator.
+		let upward_messages = if let Some(separator_index) =
+			upward_messages.iter().rposition(|message| message.is_empty())
+		{
+			upward_messages.split_at(separator_index).0
```

Comment:
This needs to also check the number of `UMPSignal` messages == 2.

Reply:
What I mean is that only 2 extra messages are added: separator and signal. Didn't go deep enough but seems if we don't enforce this, then parachains can bloat the relay chain blocks by including garbage past the ump signal. It will also have to account for the size of the encoded ump signal.